great_expectations/cli/datasource.py

```diff
@@ -103,6 +103,35 @@ def datasource_new(directory):
         sys.exit(1)
 
 
+@datasource.command(name="delete")
+@click.option(
+    '--directory',
+    '-d',
+    default=None,
+    help="Datasource to delete"
+)
+def delete_datasource(self,datasource_name=None):
+    """Delete data source"""
+    context = load_data_context_with_error_handling(directory)
+    if datasource_name is None:
+        cli_message("<red>{}</red>".format("Datasource name must be a datasource name"))
+        return
+    else:
+        datasources = context.list_datasources()
+        if not datasources.has_key(datasource_name):
```

Comment:
This must use the data_context.delete_datasource method

Reply:
We were pair walking though this. Resolved.